scripted-diff: Avoid incompatibility with CMake AUTOUIC feature

hebasto opened the pull request:
Working on [migration](https://github.com/hebasto/bitcoin/pull/3) from Autotools to CMake build system, I found that our current code base needs to be adjusted.

CMake [allows](https://cmake.org/cmake/help/latest/prop_tgt/AUTOUIC.html) to
> handle the Qt `uic` code generator automatically

When using this feature, statements like `#include "ui_<ui_base>.h"` are processed in a special way.

The `node/ui_interface.h` unintentionally breaks this feature. Of course, it is possible to provide a list of source files to be excluded from `AUTOUIC`. But, unfortunately, this approach does not work for the `qt/sendcoinsdialog.cpp` source file, where there are both https://github.com/bitcoin/bitcoin/blob/b71d37da2c8c8d2a9cef020731767a6929db54b4/src/qt/sendcoinsdialog.cpp#L10 and https://github.com/bitcoin/bitcoin/blob/b71d37da2c8c8d2a9cef020731767a6929db54b4/src/qt/sendcoinsdialog.cpp#L24

dongcarl commented:
Oh! What happens if we do `set_property(SOURCE node/ui_interface.h PROPERTY SKIP_AUTOUIC ON)`?

hebasto commented:
@dongcarl 
> Oh! What happens if we do `set_property(SOURCE node/ui_interface.h PROPERTY SKIP_AUTOUIC ON)`?

[It does not help](https://github.com/hebasto/bitcoin/commits/cmake/0611.5):
```
[ 12%] Automatic MOC and UIC for target bitcoinqt

AutoUic error
-------------
"SRC:/src/qt/sendcoinsdialog.cpp"
includes the uic file "node/ui_interface.h",
but the user interface file "interface.ui"
could not be found in the following directories
  "SRC:/src/qt/node"
  "SRC:/src/qt"
  "SRC:/src/qt/forms"
  "SRC:/src/qt/forms/node"

gmake[3]: *** [src/qt/CMakeFiles/bitcoinqt_autogen.dir/build.make:71: src/qt/CMakeFiles/bitcoinqt_autogen] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:754: src/qt/CMakeFiles/bitcoinqt_autogen.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:734: src/qt/CMakeFiles/bitcoin-qt.dir/rule] Error 2
gmake: *** [Makefile:358: bitcoin-qt] Error 2
```

hebasto commented:
Friendly ping @ryanofsky

furszy reviewed (commented):
Concept ACK

Sjors commented on src/Makefile.am:
```diff
@@ -198,7 +198,7 @@ BITCOIN_CORE_H = \
   node/psbt.h \
   node/transaction.h \
-  node/ui_interface.h \
+  node/uiinterface.h \
```
`interface_ui.h` is a bit more readable

hebasto replied:
Thanks! Updated.

hebasto committed:
scripted-diff: Avoid incompatibility with CMake AUTOUIC feature

-BEGIN VERIFY SCRIPT-
sed -i "s|node/ui_interface|node/interface_ui|g" $(git grep -l "node/ui_interface" ./src)
git mv src/node/ui_interface.cpp src/node/interface_ui.cpp
git mv src/node/ui_interface.h src/node/interface_ui.h
sed -i "s|BITCOIN_NODE_UI_INTERFACE_H|BITCOIN_NODE_INTERFACE_UI_H|g" src/node/interface_ui.h
-END VERIFY SCRIPT-

laanwj commented:
I don't know, ~0 on this, I dislike the idea of having to rename source files to make the build system happy.

> Oh! What happens if we do set_property(SOURCE node/ui_interface.h PROPERTY SKIP_AUTOUIC ON)?

I would prefer a solution like that.

MarcoFalke commented:
cr ACK 018d70b58726b361b6951e0e6de04f13eb97a89d

New name is fine and I reviewed the scripted diff.

No opinion on the build system stuff.

hebasto commented:
@laanwj 
> I don't know, ~0 on this, I dislike the idea of having to rename source files to make the build system happy.
> 
> > Oh! What happens if we do set_property(SOURCE node/ui_interface.h PROPERTY SKIP_AUTOUIC ON)?
> 
> I would prefer a solution like that.

You are right. I also prefer some kind of build system switchers. But they do not work for `qt/sendcoinsdialog.cpp` source file, which should be processed, as it contains `#include <qt/forms/ui_sendcoinsdialog.h> `, and should be skipped, as it contains `#include <node/ui_interface.h>`. Looks like it is impossible without renaming.

Sjors commented:
We already have build system code that special cases `ui_<...>.h` files, so the rename is probably a good idea anyway.

ryanofsky reviewed (approved):
Code review ACK 018d70b58726b361b6951e0e6de04f13eb97a89d

furszy reviewed (approved):
Code review ACK 018d70b5

Small scripted-diff.
